src/iop/lens.c

```diff
@@ -874,38 +872,61 @@ void modify_roi_in(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *
 
   if(modflags & (LF_MODIFY_TCA | LF_MODIFY_DISTORTION | LF_MODIFY_GEOMETRY | LF_MODIFY_SCALE))
   {
-    // acquire temp memory for distorted pixel coords
-    const size_t bufsize = (size_t)roi_in->width * 2 * 3;
-    void *const buf = dt_alloc_align(16, bufsize * dt_get_num_threads() * sizeof(float));
+    const int xoff = roi_in->x;
+    const int yoff = roi_in->y;
+    const int width = roi_in->width;
+    const int height = roi_in->height;
+    const int awidth = abs(width);
+    const int aheight = abs(height);
+    const int xstep = (width < 0) ? -1 : 1;
+    const int ystep = (height < 0) ? -1 : 1;
+
+    const size_t nbpoints = 2 * awidth + 2 * aheight;
+
+    float *buf = dt_alloc_align(16, nbpoints * 2 * 3 * sizeof(float));
+
+    float *bufptr = buf;
 
 #ifdef _OPENMP
-#pragma omp parallel default(none) shared(modifier) reduction(min : xm, ym) reduction(max : xM, yM)
+#pragma omp parallel for default(none) shared(modifier, bufptr) schedule(static)
 #endif
-    {
-      float *const bufptr = ((float *)buf) + (size_t)bufsize * dt_get_thread_num();
+    for(int i = 0; i < awidth; i++)
+      lf_modifier_apply_subpixel_geometry_distortion(modifier, xoff + i * xstep, yoff, 1, 1, bufptr + 6 * i);
+
+    bufptr += 6 * awidth;
 
 #ifdef _OPENMP
-#pragma omp for schedule(static)
+#pragma omp parallel for default(none) shared(modifier, bufptr) schedule(static)
```

Comment:
Could you please collapse it all into one `#pragma omp parallel` section with multiple `#pragma omp for` sections inside, please?
Else, i will have to do as part of upcoming cleanup.